lib/matplotlib/axes/_axes.py

```diff
@@ -5607,7 +5606,7 @@ def hist(self, x, bins=10, range=None, normed=False, weights=None,
         # basic input validation
         flat = np.ravel(x)
         if len(flat) == 0:
-            raise ValueError("x must have at least one data point")
+            return [], [], cbook.silent_list('No Patches')
```

Comment:
@efiring @mdboom @WeatherGod @pelson Thoughts, I think that this is an important API choice that needs to be made.